src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -299,38 +309,137 @@ public int bytesLeft()
     @Override
     public int write(ByteBuffer buffer) throws IOException
     {
-      WritableByteChannel channel = Channels.newChannel(out);
       return addToOffset(channel.write(buffer));
     }
 
     @Override
     public int write(InputStream in) throws IOException
     {
-      return addToOffset(Ints.checkedCast(ByteStreams.copy(in, out)));
+      return addToOffset(ByteStreams.copy(Channels.newChannel(in), channel));
```

Comment:
Yes, I've neglected this because this is not used in Druid. But seemed good this to be fixed.